polars/polars-arrow/src/utils.rs

```diff
@@ -40,3 +44,67 @@ where
         self.iter.next_back()
     }
 }
+
+fn prepare_buffers(
+    l: &Buffer,
+    offset_l: usize,
+    r: &Buffer,
+    offset_r: usize,
+    len_in_bits: usize,
+) -> (BooleanArray, BooleanArray) {
+    let l = BooleanArray::from(
+        ArrayData::builder(DataType::Boolean)
+            .add_buffer(l.clone())
+            .offset(offset_l)
+            .len(len_in_bits)
+            .build(),
+    );
+    let r = BooleanArray::from(
+        ArrayData::builder(DataType::Boolean)
+            .add_buffer(r.clone())
+            .offset(offset_r)
+            .len(len_in_bits)
+            .build(),
+    );
+    (l, r)
+}
+
+// buffer ops from arrow is private. So we hack around this by creating boolean arrays.
+// should do the same as arrows buffer_bin_and
+pub fn buffer_and(
```

Comment:
@jorgecarleitao

Is there a better way than this hack? I want to combine buffers and make sure that the offsets are taken into account.